chuffed/globals/disjunctive.cpp

```diff
@@ -246,12 +246,13 @@ class DisjunctiveEF : public Propagator {
 		}
 
 		// initialise data structures
-		ps_times = (Tint*)malloc(static_cast<size_t>(4 * x.size()) * sizeof(Tint));
-		ps_tasks = (Tint*)malloc(static_cast<size_t>(4 * x.size()) * sizeof(Tint));
-		residual = (Tint*)malloc(x.size() * sizeof(Tint));
+		unsigned tmpsize = x.size();
```

Comment:
This doesn't really solve the underlying issue that the compiler warns about.

The problem is not how the value is cast, but rather the fact that the calculation within `malloc` might go beyond the bounds of what is possible to allocate.

If we wanted to make the code-base “more correct”, then we could check the calculated size beforehand and throw an exception instead of the allocator crashing any program. However, this is mostly interesting for libraries and for Chuffed itself it will likely make very little difference.

Reply:
I agree with you; I don't understand why this suppresses the warnings. I imagine `x.size()` is small enough where it doesn't really matter. I am just trying to make my codebase totally clean of warnings.